src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -451,86 +567,91 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
-        if (!updateMatchingItinerary()) {
-            // TODO: figure out how to notify the user that the itinerary is not possible anymore
-            // for now, just say the next itinerary couldn't be found
-            return false;
+        // update overall itinerary and leg start/end times by adding offset
+        matchingItinerary.startTime = new Date(matchingItinerary.startTime.getTime() + offsetMillis);
+        matchingItinerary.endTime = new Date(matchingItinerary.endTime.getTime() + offsetMillis);
+        for (Leg leg : matchingItinerary.legs) {
+            leg.startTime = new Date(leg.startTime.getTime() + offsetMillis);
+            leg.endTime = new Date(leg.endTime.getTime() + offsetMillis);
         }
-        return true;
-    }
 
-    /**
-     * Creates and executes a request to OTP using the currently set target date and other trip plan query parameters
-     * associated with the current monitored trip. The response is saved into the {@link CheckMonitoredTrip#otpResponse}
-     * variable.
-     *
-     * @return false if any kind of exception occurred or if a trip plan could not be calculated by OTP
-     */
-    private boolean calculateOtpResponse() {
-        OtpDispatcherResponse otpDispatcherResponse;
-        try {
-            otpDispatcherResponse = OtpDispatcher.sendOtpPlanRequest(generateTripPlanQueryParams());
-        } catch (Exception e) {
-            // TODO: report bugsnag
-            LOG.error("Encountered an error while making a request ot the OTP server. error={}", e);
-            return false;
+        // update journey state with baseline departure and arrival times which are the last known departure/arrival
+        journeyState.baselineDepartureTimeEpochMillis = matchingItinerary.startTime.getTime();
+        journeyState.baselineArrivalTimeEpochMillis = matchingItinerary.endTime.getTime();
+
+        // update journey state with the original (scheduled departure and arrival times). Calculate these by
+        // finding the first/last transit legs and subtracting any delay.
+        journeyState.scheduledDepartureTimeEpochMillis = matchingItinerary.startTime.getTime();
+        for (Leg leg : matchingItinerary.legs) {
+            if (leg.transitLeg) {
+                journeyState.scheduledDepartureTimeEpochMillis -= TimeUnit.MILLISECONDS.convert(
+                    leg.departureDelay,
+                    TimeUnit.SECONDS
+                );
+                break;
+            }
+        }
+        journeyState.scheduledArrivalTimeEpochMillis = matchingItinerary.endTime.getTime();
+        for (int i = matchingItinerary.legs.size() - 1; i >= 0; i--) {
+            Leg leg = matchingItinerary.legs.get(i);
+            if (leg.transitLeg) {
+                journeyState.scheduledArrivalTimeEpochMillis -= TimeUnit.MILLISECONDS.convert(
+                    leg.arrivalDelay,
+                    TimeUnit.SECONDS
+                );
+                break;
+            }
         }
```

Comment:
This seems very familiar to `Itinerary#getScheduledEndTimeEpochMillis`?

Reply:
Not sure why I didn't use that. Fixed in 1987213b6754ffa98f4748820b628be64f9443f4